### DISCUSSION TITLE
Only issues in the project board?

### DISCUSSION THREAD

**Discussion by obulat** [WordPress/openverse#11] [Ideas]
I feel like it would be easier to understand the overall picture of what's going on in the project if we only add issues, and not add PRs to it. Every issue already has an indication if it has a linked PR or not, and a link to that PR, if necessary. 

When both an issue and a PR for it are in the board separately, it feels more crowded. And there's potentially some chance of forgetting to update one after updating the other, so the statuses of essentially the same thing will be different.

**Comment by zackkrida** [comment]
I agree with this. I think the only reason to have PRs on a project board is for PRs that do not have a linked issue. But we can probably figure out a solution for that problem.

**Reply by zackkrida** [reply]
Oh, and also the case of multiple PRs for a single issue. If the linked PRs are in different phases of review it might be beneficial to have them on the board.

**Comment by dhruvkb** [comment]
Having PRs in the project board does have it's advantages too, as it helps to visualise what stage of the review process the PR currently is, so that it can be appropriately nudged ahead towards either merge or closure.

**Comment by zackkrida** [comment]
Revisiting this, I'm thinking the PRs add a lot of clutter, and any PR being tracked _should_ be linked to its issue anyway. Curious what y'all think @WordPress/openverse-developers

**Reply by dhruvkb** [reply]
**Pros:**
I still stand by my [original argument](https://github.com/WordPress/openverse/discussions/11#discussioncomment-671536) that I find it very helpful to see PRs in there. For example I routinely check the board to see which PRs are in 'Review required' so that I can review them if I have some time. It's also a quick way to see if my any of my PRs are 'Reviewer approved' and ready to be merged. As PRs pile up in 'Review required', it's also a constant reminder that reviews are just as important as dev work.

**Cons:**
The clutter I believe is only in the 'In progress' column where linked issues and WIP PRs are placed together. But then again, if something is in progress, it neither needs someone's review and you probably will continue to work on it on your own so the board isn't supposed to help there. It's the column that you're least likely to look at.

**Ideology:**
Finally, as a person who routinely makes small PRs without issues when the motive is clear (sorry, I'm lazy), I heavily feel that a project's true status is reflected by the progress on both issues and PRs. Not having PRs on the board means having to look across 5 repos (and probably in notifications, which is bad!) to find review requests and approved PRs.

**Compromise:**
As a compromise between clutter and incomplete information, might I suggest breaking down the columns so that PRs and issues are never in one column and moving PR columns to end of the board?

**Automation:**
As for the issue of managing state for issues and PRs that @obulat raised in the original discussion, it's almost 100% automated anyway. Both issues and PRs move themselves around and keep themselves updated. The only manual work is moving issues from 'Backlog' to 'To do' (which you can skip by the way, they'll move to 'In progress' when a PR is linked to them) or 'Blocked' (which can be automated but just hasn't yet because that's infrequent).

**Reply by AetherUnbound** [reply]
From what I can tell reading everyone's replies, it seems like the primary value-add of having PRs in the board is that it's easy to see which ones need review. For myself, I've found that the [GitHub Pull Requests](https://github.com/pulls) page (specifically the [Review Requested tab](https://github.com/pulls/review-requested)) is a great place for at-a-glance views of what still needs a review. Do we think that could be a sufficient substitute, or at least one that would allow us to only have issues on the project board?

**Comment by krysal** [comment]
I have been thinking about this the last few days and though I agree the board feels a little cluttered with both issues and PRs, I don’t have strong reasons to remove or keep the PRs.

However, this is closely related to if we should make a requirement that every PR has a linked issue, and I believe we should adopt this, it’ll give more formality to the process, to open the conversation to identify the problem clearly (if there is one) or the improvement, and its possible solutions and timing/priority. 

This decision to answer this question will depend on what do we want to use this dashboard for? If it’s to have a general strategic view of the Openverse projects, I’ll say the PRs are in excess, but again I don’t mind keeping them. I find the filters of the GH notification UI useful; you can reduce the list by repository, if you are requested for review, if you’re assigned, and more; it still requires some maintenance work though, so I’m curious to know how the rest (want to) use this dashboard.

**Comment by zackkrida** [comment]
I am strongly leaning towards:

- More strict enforcement of issue creation + linked issues
- Removing PRs from the Project Board

The only reason I see _not_ to do this—it might be difficult to move _issues_ to the 'needs review' column automatically. @dhruvkb do you have a sense of if that is even possible?

**Reply by dhruvkb** [reply]
Given enough time and effort, anything is possible. Wisdom aside, yeah that's feasible if we all agree that it would be a beneficial thing to do.

**Reply by zackkrida** [reply]
@dhruvkb I suppose I meant to ask if it was reasonable to do so

**Reply by dhruvkb** [reply]
It is reasonable. Having fewer cards on the board could make the board much more readable. What's the team's opinion on a separate board (or completely separate columns) just for PRs?

**Reply by AetherUnbound** [reply]
I really like that idea! One for planning and one for work/reviewing

**Reply by zackkrida** [reply]
I like the idea of a PR-only board! It would serve as a good test too—we can see which one folks end up using the most and how it impacts our workflows.

**Reply by dhruvkb** [reply]
Made some progress in this area:

- [Openverse PRs board](https://github.com/orgs/WordPress/projects/8)
- [PR to update automations](https://github.com/WordPress/openverse/pull/84)

**Comment by zackkrida** [comment]
This was fixed via dedicated issue and PR boards.
